src/coll/options.rs

```diff
@@ -89,28 +171,134 @@ pub struct FindOptions {
     pub read_preference: Option<ReadPreference>,
 }
 
+impl FindOptions {
+    /// Creates a new FindOptions struct with default parameters.
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl From<FindOptions> for bson::Document {
+    fn from(options: FindOptions) -> Self {
+        let mut document = bson::Document::new();
+
+        // `allow_partial_results`, `no_cursor_timeout`, `oplog_relay`, and `cursor_type` are used by
+        // wire_protocol::OpQueryFlags.
+        //
+        // `max_time_ms` and `modifiers` are not currently used by the driver.
+        //
+        // read_preference is used directly by Collection::find_with_command_type.
+
+        if let Some(projection) = options.projection {
+            document.insert("projection", Bson::Document(projection));
+        }
+
+        if let Some(skip) = options.skip {
+            document.insert("skip", Bson::I64(skip));
+        }
+
+        if let Some(limit) = options.limit {
+            document.insert("limit", Bson::I64(limit));
+        }
+
+        if let Some(batch_size) = options.batch_size {
+            document.insert("batchSize", Bson::I32(batch_size));
+        }
+
+        if let Some(sort) = options.sort {
+            document.insert("sort", Bson::Document(sort));
+        }
+
+        document
+    }
+}
+
 /// Options for `findOneAndDelete` operations.
-#[derive(Clone, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct FindOneAndDeleteOptions {
     pub max_time_ms: Option<i64>,
     pub projection: Option<bson::Document>,
     pub sort: Option<bson::Document>,
     pub write_concern: Option<WriteConcern>,
 }
 
+impl FindOneAndDeleteOptions {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl From<FindOneAndDeleteOptions> for bson::Document {
+    fn from(options: FindOneAndDeleteOptions) -> Self {
+        let mut document = bson::Document::new();
+
+        // max_time_ms is not currently used by the driver
+
+        if let Some(projection) = options.projection {
+            document.insert("fields", Bson::Document(projection));
+        }
+
+        if let Some(sort) = options.sort {
+            document.insert("sort", Bson::Document(sort));
+        }
+
+        if let Some(write_concern) = options.write_concern {
+            document.insert("writeConcern", Bson::Document(write_concern.to_bson()));
+        }
+
+        return document;
+    }
+}
+
 /// Options for `findOneAndUpdate` operations.
-#[derive(Clone)]
+#[derive(Clone, Debug, Default)]
 pub struct FindOneAndUpdateOptions {
-    pub return_document: ReturnDocument,
+    pub return_document: Option<ReturnDocument>,
     pub max_time_ms: Option<i64>,
     pub projection: Option<bson::Document>,
     pub sort: Option<bson::Document>,
-    pub upsert: bool,
+    pub upsert: Option<bool>,
     pub write_concern: Option<WriteConcern>,
 }
 
+impl FindOneAndUpdateOptions {
+    pub fn new() -> Self {
+        Default::default()
```

Comment:
From what I've read, it's considered more idiomatic to implement `new` by calling `default` rather than vice-versa, presumably because `Default` is often derived rather than directly implemented.